### DISCUSSION TITLE
There are some questions about layout&&use client

### DISCUSSION THREAD

**Discussion by yanqc1996** [vercel/next.js#54917] [App Router]
I upgraded to next.js 13.4 in my project, and it is not as useful as expected. The effect I want to achieve on one of my pages is: the navbar needs to be transparent initially (the others are normal display colors), but if I scroll over the head module or hover, the navbar will return to the normal color, which leads me to use a variable to controls the display of the navbar. I tried to use context to achieve it, but at this time I had to set it to use client, which caused my change logic to be executed on the client side, causing my page to first display normal colors and then flash to transparent, which is not useful for the first screen has caused a very bad effect. I would like to ask if there is a better way to achieve this function?

**Comment by devjiwonchoi** [comment]
If the feature you need is an initial transparent nav to color on hover, don't think you need to go over using `useContext`, and it seems like more of a styling issue, right?

**Reply by yanqc1996** [reply]
No, what I want to achieve is two different navbars, transparent and normal, of course this can be achieved by using different layout files, but when I want to achieve it with one layout file, there will be problems, I cannot set the initial parameters To change the display of the layout (when I use context, the initial processing has to be done on the client side). In addition, another problem also bothers me. When I rely on a client component such as swiper in the hero header component, the response of the client is always slow. I am not sure whether it is because of the unified processing of the client method, but responses like the first screen will make people feel uncomfortable, and I would like to have a way to quickly process some component clients first. Like this page https://vibe-portal-web-git-staging-vibeus.vercel.app/apps/vibe-rooms/ for example, when on the mobile phone, the effect of the first screen is really bad
